tap_kafka/sync.py

```diff
@@ -69,40 +69,40 @@ def init_local_store(kafka_config):
 
 def init_kafka_consumer(kafka_config):
     LOGGER.info('Initialising Kafka Consumer...')
-    return KafkaConsumer(
+    consumer = Consumer({
         # Required parameters
-        kafka_config['topic'],
-        bootstrap_servers=kafka_config['bootstrap_servers'],
-        group_id=kafka_config['group_id'],
+        'bootstrap.servers': kafka_config['bootstrap_servers'],
+        'group.id': kafka_config['group_id'],
 
         # Optional parameters
-        consumer_timeout_ms=kafka_config['consumer_timeout_ms'],
-        session_timeout_ms=kafka_config['session_timeout_ms'],
-        heartbeat_interval_ms=kafka_config['heartbeat_interval_ms'],
-        max_poll_records=kafka_config['max_poll_records'],
-        max_poll_interval_ms=kafka_config['max_poll_interval_ms'],
+        'session.timeout.ms': kafka_config['session_timeout_ms'],
+        'heartbeat.interval.ms': kafka_config['heartbeat_interval_ms'],
+        'max.poll.interval.ms': kafka_config['max_poll_interval_ms'],
 
         # Non-configurable parameters
-        enable_auto_commit=False,
-        auto_offset_reset='earliest',
-        value_deserializer=lambda m: json.loads(m.decode(kafka_config['encoding'])))
+        'enable.auto.commit': False,
+        'auto.offset.reset': 'earliest',
+    })
+    consumer.subscribe([kafka_config['topic']])
+
+    return consumer
 
 
 def kafka_message_to_singer_record(message, topic, primary_keys):
     """Transforms kafka message to singer record message"""
     # Create dictionary with base attributes
     record = {
-        "message": message.value,
-        "message_timestamp": message.timestamp,
-        "message_offset": message.offset,
-        "message_partition": message.partition
+        "message": message.value(),
+        "message_timestamp": message.timestamp(),
+        "message_offset": message.offset(),
+        "message_partition": message.partition()
     }
 
     # Add primary keys to the record message
     for key in primary_keys:
         pk_selector = primary_keys[key]
         try:
-            record[key] = dpath.util.get(message.value, pk_selector)
+            record[key] = dpath.util.get(message.value(), pk_selector)
```

Comment:
I noticed that the code continues executing if PK is not found in the message which i think is not the right behavior, it also makes the target fail.

I know this isn't related to this PR, so perhaps we could have a ticket to update/discuss this behavior.

Reply:
Thanks, I agree, we need to agree what's the expected behaviour and address separately out of this PR.